packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -372,6 +256,21 @@ export interface UserPoolProps {
    */
   readonly autoVerify?: AutoVerifiedAttrs;
 
+  /**
+   * The set of attributes that are required for every user in the user pool.
+   * Read more on attributes here - https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html
+   *
+   * @default - No attributes are required.
+   */
+  readonly requiredAttributes?: StandardAttribute[];
```

Comment:
Why not a boolean map?

```ts
requiredAttributes: {
  email: true,
  phone: true
}
```

Reply:
I don't see much difference between having a boolean map vs. a list of enum here. Both provide pretty much a very similar experience.

The boolean map is quite handy when it comes to settings, such as "enable xxx" and there are some that default to `true` and some to `false`. 
OTOH over here, we're asking the user to specify a set of fields that they want to make mandatory, and an array makes good sense.
Both have the same level of IDE autocomplete support.

Implementation wise - mapping each entry from a boolean map to the correct attribute name is another moving part that introduces human error.
I understand this needs to be designed from the customer experience and not from implementation complexity, but when the two experiences are close, it's worth considering this.

I'd prefer to leave this as a list of `StandardAttribute`. How strongly do you feel about having a boolean map here?

Reply:
> Both have the same level of IDE autocomplete support.

I disagree that the ergonomics are equivalent. The IDE won’t expand the enum automatically and also will allow the user to specify each value multiple times which doesn’t make sense. 

An array is not the “correct” data type here in my opinion, but I don’t think it’s critical!

Reply:
Fair enough. I've modified this per your suggestion.